java-generator/it/src/test/java/io/fabric8/java/generator/gradle/plugin/ITGradleRunnerExtension.java

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.java.generator.gradle.plugin;
+
+import org.gradle.testkit.runner.BuildResult;
+import org.gradle.testkit.runner.GradleRunner;
+import org.junit.jupiter.api.extension.AfterEachCallback;
+import org.junit.jupiter.api.extension.BeforeEachCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+import java.io.File;
+import java.net.URI;
+import java.util.Arrays;
+import java.util.stream.Collectors;
+
+/**
+ * Simplified/updated version of
+ * https://github.com/eclipse/jkube/blob/master/gradle-plugin/it/src/test/java/org/eclipse/jkube/gradle/plugin/tests/ITGradleRunnerExtension.java
+ */
+public class ITGradleRunnerExtension implements BeforeEachCallback, AfterEachCallback {
+  private GradleRunner gradleRunner;
+
+  @Override
+  public void beforeEach(ExtensionContext context) throws Exception {
+    gradleRunner = GradleRunner.create()
+        .withGradleDistribution(new URI("https://services.gradle.org/distributions/gradle-8.1-bin.zip"))
```

Comment:
would it be possible to bind the gradle version to a Maven property so that it will be updated by dependabot?

Reply:
Yep. I think this is doable by adding a `*.properties` file that binds a property value to one defined by the Maven project that's executing the tests.
BTW - I must say this would diverge from the version of the Gradle artifacts we're using, which is behind ATM (see https://github.com/fabiobrz/kubernetes-client/blob/java-gen.gradle-plugin/java-generator/pom.xml#L43) and which doesn't seemm to be actively released lately, see e.g. `gradle-core` [1].
WDYT?

[1]
https://mvnrepository.com/artifact/org.gradle/gradle-core?repo=gradle-libs-releases-local

Reply:
You should be able to skip the `properties` file and directly generate from Maaven a Java source file containing the version, we usually use something like this: https://stackoverflow.com/a/36628755

Regarding the gradle versions issues, I'm not really familiar with them ... maybe @manusa ?

Reply:
Should be fixed now. Currently the Gradle binaries version is set by default to `8.1`. Can be overridden via `-Dgradle-binaries.version=...`

Reply:
perfect 👍